dash/backend/src/modules/falco/dao/falco.dao.ts

```diff
@@ -368,30 +368,147 @@ export class FalcoDao {
         }
     }
 
-    async createFalcoRule(rule: FalcoRuleDto): Promise<FalcoRuleDto> {
-        const raw = instanceToPlain(rule);
+    async createFalcoRule(rule: FalcoRuleCreateDto): Promise<number> {
         const knex = await this.databaseService.getConnection();
-        return knex.into('falco_rules')
-          .insert(raw, '*')
-          .then(resp => plainToInstance(FalcoRuleDto, resp[0]));
+        return knex.transaction(async (trx) => {
+            rule.allNamespaces = !rule?.namespaces?.length;
+            rule.allClusters = !rule?.clusters?.length;
+            let rawNamespaces = [];
+            if (!rule.allNamespaces) {
+                rawNamespaces = rule.namespaces.map(ns => {
+                    return { namespace: ns }
+                })
+            }
+            let rawClusters = [];
+            if (!rule.allClusters) {
+                rawClusters = rule.clusters.map(id => {
```

Comment:
id => ({ cluster_id: id })

Reply:
Latest commit uses this syntax for all 4 maps in create & update dao functions